### PR TITLE
Features/initial implementation

### PR DESCRIPTION
**Pull request by Peiling-Ding**

### CODE REVIEW

**Comment by StarWindMoonCloud** [comment]
This plugin is highly coupled with manhattan_config repo, a more specific repo name is recommended

File: src/main/java/com/pmi/SchemaCompiler/SchemaGeneratorMojo.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  // The resource directory which should contain a schema folder having main.yml inside it.
+  @Parameter(defaultValue = "${project.build.resources[0].directory}", required = true)
+  private File resourceDir;
+
+  @Parameter(defaultValue = "${project.build.sourceDirectory}", required = true)
+  private File sourceDir;
+
+  public void execute() throws MojoExecutionException {
+    try {
+      Path schemaDirPath = Paths.get(resourceDir.getPath(), "schema");
```
Comment (StarWindMoonCloud) [inline src/main/java/com/pmi/SchemaCompiler/SchemaGeneratorMojo.java:37]: this is highly coupled, can we pass a parameter?

File: src/main/java/com/pmi/SchemaCompiler/SchemaValidatorMojo.java [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  private void validate(ObjectMapper objectMapper, Path filePath, Class targetC) throws Exception {
+    if (filePath.getFileName().toString().endsWith(".json")) {
+      getLog().info(MessageFormat.format("Validating file: {0}", filePath.toString()));
+      Object targetObject =
+          objectMapper
+              .disable(DeserializationFeature.ACCEPT_FLOAT_AS_INT)
+              .enable(DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES)
+              .enable(DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY)
+              .readValue(filePath.toUri().toURL(), targetC);
+      JsonNode acutal = objectMapper.readTree(objectMapper.writeValueAsString(targetObject));
```
Comment (StarWindMoonCloud) [inline src/main/java/com/pmi/SchemaCompiler/SchemaValidatorMojo.java:123]: expected and actual should switch

File: src/main/java/com/pmi/SchemaCompiler/SchemaGeneratorMojo.java (hunk 1 above)
Comment (Peiling-Ding) [reply src/main/java/com/pmi/SchemaCompiler/SchemaGeneratorMojo.java:37]: Yes, we can mark `schemaDirPath` as a parameter.

File: src/main/java/com/pmi/SchemaCompiler/SchemaValidatorMojo.java (hunk 2 above)
Comment (Peiling-Ding) [reply src/main/java/com/pmi/SchemaCompiler/SchemaValidatorMojo.java:128]: Fixed

**Pull request merged**